### ISSUE TITLE
rocker/binder basic example gives 'jupyter-notebook' not found error

### ISSUE THREAD

**Issue by fmannhardt** [rocker-org/rocker-versioned2#147]
I am trying to run the simple example provided on this page:
https://hub.docker.com/r/rocker/binder

It gives the following error:
```
> docker run -p 8888:8888 rocker/binder
Unable to find image 'rocker/binder:latest' locally
latest: Pulling from rocker/binder
a70d879fa598: Already exists
c4394a92d1f8: Already exists
10e6159c56c0: Already exists
833bebb909e4: Pull complete
9bceb8b50184: Pull complete
e925a327b30d: Pull complete
01c3425e0ade: Pull complete
aa899054de42: Pull complete
3073722381fa: Pull complete
c11e2f8a575b: Pull complete
1b74c9d06d50: Pull complete
5081f19c4c6e: Pull complete
Digest: sha256:9ea13371c43d83c161da413669846eb5d19e67e7245686d2926c99ef99e07849
Status: Downloaded newer image for rocker/binder:latest
Traceback (most recent call last):
  File "/usr/local/bin/jupyter", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/jupyter_core/command.py", line 285, in main
    command = _jupyter_abspath(subcommand)
  File "/usr/local/lib/python3.8/dist-packages/jupyter_core/command.py", line 124, in _jupyter_abspath
    raise Exception(
Exception: Jupyter command `jupyter-notebook` not found.
```

**Comment by eitsupi** [comment]
It may be due to the fact that `jupyter-server-proxy`, which is required when installing `jupyter-rsession-proxy`, no longer installs `notebook`.
https://github.com/jupyterhub/jupyter-server-proxy/commit/48eec7295499ce6a0e72348a16fd77146cae9d4d

**Comment by fmannhardt** [comment]
I can confirm that it is fixed by changing:
`pip3 install --no-cache-dir jupyter-rsession-proxy`
to:
`pip3 install --no-cache-dir jupyter-rsession-proxy jupyterlab~=3.0 notebook`
here:
https://github.com/rocker-org/rocker-versioned2/blob/96d41e95c5e83f8d388a4116cb0df1c0bd83313c/scripts/install_binder.sh#L24

I still don't get a R Kernel in the Jupyter Notebook though and a error `500 : Internal Server Error` when trying to access RStudio, Maybe related to this issue here:
https://github.com/rocker-org/rocker-versioned2/issues/88 

(This is all using 4.x R)

**Comment by fmannhardt** [comment]
Something is wrong in the way the `IRkernel::installspec` call is done. I simply added (to the end of my Dockerfile:
```
RUN R --quiet -e "IRkernel::installspec()"
```
and now the R Kernel shows up and works.

**Comment by fmannhardt** [comment]
Starting RStudio the errors look like:

```
19 Apr 2021 11:27:16 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:16 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:16 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:16 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:16 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:16 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:16 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:16 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:16 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:17 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:17 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:17 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:17 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:17 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:17 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:17 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:17 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:17 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:17 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:17 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:17 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:18 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:18 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:18 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:18 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:18 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:18 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:18 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:18 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:18 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:19 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:19 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:19 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:19 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:19 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:19 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:19 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:19 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:19 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:19 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:19 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:19 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:20 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:20 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:20 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:20 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:20 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:20 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
19 Apr 2021 11:27:20 [rserver] ERROR system error 1 (Operation not permitted) [path: /var/run/rstudio-server/revocation-list, description: Could not set revocation file permissions to 600 for file: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::system::changeFileMode(const rstudio::core::FilePath&, mode_t) src/cpp/server/ServerMain.cpp:62; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:531
19 Apr 2021 11:27:20 [rserver] ERROR Could not read revocation list; LOGGED FROM: rstudio::core::Error rstudio::server::auth::handler::initialize() src/cpp/server/auth/ServerAuthHandler.cpp:539
19 Apr 2021 11:27:20 [rserver] ERROR system error 2 (No such file or directory) [path: /var/run/rstudio-server/revocation-list]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1150; LOGGED FROM: int main(int, char* const*) src/cpp/server/ServerMain.cpp:674
[W 11:27:21.056 NotebookApp] 500 GET /rstudio/ (172.17.0.1): could not start rstudio in time
```

Seems like a wrong permissions issue?

**Comment by mark-andrews** [comment]
Following @fmannhardt's advice, I dealt with the `Jupyter command `jupyter-notebook` not found.` issue, and got a Jupyter session working, but when i try to launch RStudio, I encountered the same Rstudio server error as described by @fmannhardt.

Does anyone have any advice on how to deal with that problem?

**Comment by Robinlovelace** [comment]
I have also noticed this issue. Very keen to see discussion of this, there have been issues with RStudio/Binder for ages and I think rocker-versioned2 could be the ideal place to address some of them with solutions that benefit many people.

**Comment by eitsupi** [comment]
Handling pip packages has been a problem for a long time, but now that [the wiki](https://github.com/rocker-org/rocker-versioned2/wiki) records the installed pip packages for each image, I think it's possible to make destructive changes now.
I think it's a good idea to add the missing packages to `install_binder.sh` as a first step.

**Comment by eitsupi** [comment]
This issue seems resolved by #309.
`rocker/binder:latest` seems work correctly.
https://github.com/rocker-org/rocker-versioned2/wiki/binder_6fac4d6e57c3

**Comment by cboettig** [comment]
:tada: :tada: 

Thanks @yuvipanda, @ryanlovett and @eitsupi for all the work here.  Very nice to have this fixed, updated in the build scripts and the fixed version automatically deployed.  :rocket:

**Comment by yuvipanda** [comment]
Yay! Thanks for the merge, @cboettig! Can this issue be closed now?

**Comment by cboettig** [comment]
probably, I was just waiting to confirm on my end at https://mybinder.org/v2/gh/rocker-org/binder/master?urlpath=rstudio but currently keep seeing timeouts there... ?

**Comment by ryanlovett** [comment]
I noticed something similar when I had tried rocker/binder + notebook a week or so ago. I don't think it is binder infrastructure. There was a javascript console error indicating something wrong with an HTTP header, so I suspected one of the jupyter proxy layers. However the proxies worked outside of rocker in a more minimal image. In any case, I still haven't a chance to explore further.

**Comment by yuvipanda** [comment]
https://github.com/yuvipanda/rstudio-binder-template works fine for me. I'm 100% convinced that is rocker-org/binder failing because the image was already built, and hence used the older, non-working `rocker/binder:latest` image. Maybe we can switch to a tag?

**Comment by eitsupi** [comment]
I don't really understand how mybinder works, but comparing https://github.com/yuvipanda/rstudio-binder-template/tree/411ca6a00fc2bd2ed390e23f8bdf439f4d28c128 and https://github.com/rocker-org/binder/tree/82a50434cddf87eb78ec20279120076ed4763be2, the path to the Dockerfile is different.
Is there a possible impact from this?

I did not see any option to set the path to the Dockerfile when creating the URL.
https://mybinder.org/

**Comment by eitsupi** [comment]
I tried playing with my fork, but I could not get RStudio Server to start...

**Comment by yuvipanda** [comment]
I opened https://github.com/rocker-org/binder/pull/46 to fix [that repo](https://github.com/rocker-org/binder/). However, newer versions of the binder image are built from *this* repo, and don't need anything to fix. Maybe the rocker/binder repo should be converted into an example instead? Maybe https://github.com/yuvipanda/rocker-binder-template can be merged into it?

**Comment by cboettig** [comment]
@yuvipanda yes, that's a great idea! Do you want to send a PR for that?

**Comment by eitsupi** [comment]
I think this issue has been resolved by #309, so I close it.
